app/models/course/level.rb

```diff
@@ -1,4 +1,15 @@
 class Course::Level < ActiveRecord::Base
   validates :experience_points_threshold, numericality: { greater_than: 0 }
   belongs_to :course, inverse_of: :levels
+  default_scope { order(:experience_points_threshold) }
+  attr_writer :level_number
+
+  # This accessor aims to prevent a potentially hard-to-find bug from hiding.
```

Comment:
I think the language is off -- 'Retrieves the level number of the current level, relative to the other levels in the same course.'

Then specify what circumstances it raises an exception.